src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+```
+
+```json title="Response"
+[
+	{
+		results: {
+			score: [
+				76,
+				83,
+				69,
+				73
+			]
+		}
+	}
+]
+```
+
+This idiom selects all elements in the `score` array.
+
+### Last Element
+
+Addionally to access the last element of an array, use `[$]`. Refereing to the `student` record above, we can access the last element of the `results` array using the following idiom:
+
+```surql
+SELECT results[$].score FROM student;
+```
+```json title="Response"
+[
+	{
+		results: {
+			score: 73
+		}
+	}
+]
+```
+
+This idiom accesses the last element of the `score` array.
+
+### First Element
+
+As we saw initially to access the first element of an array, use `[0]`.
+
+```surql
+SELECT results[0].score FROM person;
+```
+
+```json title="Response"
+[
+  {
+    "results": [
+      { "score": 76 }
+    ]
+  }
+]
+```
+
+### Flatten
+
+To flatten nested arrays, use `...` (ellipsis). This is useful when you want to access all the elements in a nested array. 
+
+```surql
+SELECT results.score... FROM student;
+```
+
+This idiom flattens any nested arrays within `results.score`, returning a single-level array.
+
+### Where Filter
+
+You can also filter elements based on a condition, use `[WHERE ... ]`. This will only return the elements that match the condition.
+
+```surql
+SELECT results[WHERE score > 70].score FROM student;
+```
+
+```json title="Response"
+[
+	{
+		results: {
+			score: [
+				76,
+				83,
+				73
+			]
+		}
+	}
+]
+```
+This idiom selects elements in `results.score` where the `score` is greater than 70.
+
+### Method chaining
+
+To call a method on the current data, use a dot `.` followed by the method name and parentheses `()` with arguments. SurrealDB supports method chaining, so you can call multiple methods(functions) on the same data. Learn more about [method chaining](/docs/surrealql/functions/database#method-syntax) in the functions section.
+
+For example, let's create a new `person` record 
+
+```surql title="Create a new person record"
+CREATE person CONTENT {
+    name: "John Doe",
+    age: 30,
+    address: {
+      city: "New York",
+      country: "USA"
+    }
+};
+```
+
+```json title="Response"
+[
+  {
+    "person": {
+      "name": "John Doe",
+      "age": 30,
+      "address": {
+        "city": "New York",
+        "country": "USA"
+      }
+    }
+  }
+]
+```
+
+To call the `uppercase()` method on the `name` field, you can use the following idiom:
+
+```surql
+SELECT name.uppercase() FROM person;
+```
+
+```json title="Response"
+[
+  {
+    "name": "JOHN DOE"
+  }
+]
+```
+
+In the example above, `uppercase()` is a method called on `person.name` to convert it to uppercase. Although this method is called as `.uppercase()` it is actually the [`string::uppercase()`](/docs/surrealql/functions/database/string#stringuppercase) function that is called. 
+
+SurrealDB will automatically convert the idiom part `.uppercase()` to the function `string::uppercase()` when the query is executed.
```

Comment:
I think rather than 'convert' you could say that SurrealDB ends up calling the regular function in every case. Maybe something like this:

SurrealDB will automatically recognize that .uppercase() refers to the function `string::uppercase` and call this function when the query is executed.